core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestUnwrapSingleColumnRowInApply.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.spi.type.RowType;
+import io.trino.sql.parser.SqlParser;
+import io.trino.sql.planner.TypeAnalyzer;
+import io.trino.sql.planner.assertions.ExpressionMatcher;
+import io.trino.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.trino.sql.planner.plan.Assignments;
+import io.trino.sql.tree.InPredicate;
+import io.trino.sql.tree.QuantifiedComparisonExpression;
+import io.trino.sql.tree.SymbolReference;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.apply;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.expression;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.project;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.sql.tree.ComparisonExpression.Operator.EQUAL;
+import static io.trino.sql.tree.QuantifiedComparisonExpression.Quantifier.ALL;
+import static java.util.Collections.emptyList;
+
+public class TestUnwrapSingleColumnRowInApply
```

Comment:
Optional suggestion: we could also add a test in `TestHiveProjectionPushdownIntoTableScan` which would also verify that the dereferenced field after unwrap is pushed into table scan.

Also, I'm not sure if there's already a query in our end-to-end tests which would use IN/QuantifiedComparisonExpression with a row type. If not, it'd be good to add that instead of the hive pushdown plan test.

Reply:
Added tests